### PR TITLE
[DYN-3710] Enable Show CodeBlockNode Lines Number

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Add and enable a toggle on the Preferences modal to turn CodeBlockNode Lines Number on and off
image

codeblocklines2

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )

### FYIs

Roberto Tellez (@RobertGlobant20 )

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@Astul-Betizagasti - if you check Dynamo 2.10 - when this setting is disabled, existing codeblocks should be modified, you should not need to recreate them to have the line numbers become visible or invisible.

**Comment by Astul-Betizagasti** [comment]
> @Astul-Betizagasti - if you check Dynamo 2.10 - when this setting is disabled, existing codeblocks should be modified, you should not need to recreate them to have the line numbers become visible or invisible.

I just added a commit to fix this issue and updated the gif on the description to reflect the right behavior

**Comment by Astul-Betizagasti** [comment]
I found a regression regarding the Saved Changes label on the Preferences modal and included a fix on this PR.
Now that the PreferencesViewModel is being saved on DynamoViewModel, instead of creating a new one every time the model opens, the Saved Changes label and its corresponding tooltip must be cleared every time the model is opened

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs [hunk 1]
```diff
@@ -641,8 +656,6 @@ public PreferencesViewModel(DynamoViewModel dynamoViewModel)
             SavedChangesLabel = string.Empty;
             SavedChangesTooltip = string.Empty;
 
-            this.PropertyChanged += Model_PropertyChanged;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:644]: Any reason why we delete this line?

Comment (Astul-Betizagasti) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:644]: yes, it was duplicated on line 650.

Comment (Astul-Betizagasti) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:644]: yes, it was duplicated in line 650

Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:644]: I see, thanks @Astul-Betizagasti

**Review by QilongTang — APPROVED** [review]

**Pull request merged**